ci-operator/step-registry/ipi/conf/azure/ipi-conf-azure-commands.sh

```diff
@@ -20,41 +20,131 @@ function getVersion() {
   echo "${version}"
 }
 
+function join_by { local IFS="$1"; shift; echo "$*"; }
+
+# BootstrapInstanceType gets its value from pkg/types/azure/defaults/platform.go
+architecture=${OCP_ARCH:-"amd64"}
+if [[ "${CLUSTER_TYPE}" == "azure-arm64" ]]; then
+  architecture="arm64"
+fi
+
 CONFIG="${SHARED_DIR}/install-config.yaml"
 
 REGION="${LEASED_RESOURCE}"
 echo "Azure region: ${REGION}"
 
+if [[ "${COMPUTE_NODE_TYPE}" == "" ]]; then
+  if [ "${architecture}" = "arm64" ]; then
+    COMPUTE_NODE_TYPE="Standard_D4ps_v5"
+  else
+    COMPUTE_NODE_TYPE="Standard_D4s_v3"
+  fi
+fi
 workers=3
 if [[ "${SIZE_VARIANT}" == "compact" ]]; then
   workers=0
 fi
 master_type=null
-if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
-  master_type=Standard_D32s_v3
-elif [[ "${SIZE_VARIANT}" == "large" ]]; then
-  master_type=Standard_D16s_v3
-elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
-  master_type=Standard_D8s_v3
+if [ "${architecture}" = "arm64" ]; then
+  BOOTSTRAP_NODE_TYPE="Standard_D8ps_v5"
+  if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+    master_type="Standard_D32ps_v5"
+  elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+    master_type="Standard_D16ps_v5"
+  elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+    master_type="Standard_D8ps_v5"
+  fi
+else
+  BOOTSTRAP_NODE_TYPE="Standard_D8s_v3"
+  if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+    master_type="Standard_D32s_v3"
+  elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+    master_type="Standard_D16s_v3"
+  elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+    master_type="Standard_D8s_v3"
+  fi
+fi
+
+echo "Using control plane instance type: ${master_type}"
+echo "Using compute instance type: ${COMPUTE_NODE_TYPE}"
+
+# set the parameters we'll need as env vars
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+
+# log in with az
+if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
+    az cloud set --name AzureUSGovernment
+else
+    az cloud set --name AzureCloud
+fi
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+# Generate availability zones with OpenShift Installer required instance types
```

Comment:
Do you think is possible to minimize the branches in the following if conditionals?

Reply:
for this part, I refer to [ipi-conf-aws-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh#L116-L130) , I think it will consider all the situations.